### PR TITLE
[C# and Python APIs] Expose knobs to enable/disable platform telemetry collection

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: Currently, official ORT packages for Windows (using TraceLogging) are configured to collect telemetry. The C API exposes knobs to disable this but C# and Python do not. This change addresses that.

**Motivation and Context**
1) Minimize gap between C#/Python and the C APIs
2) Account for user request in https://github.com/microsoft/onnxruntime/issues/3898 (`do I have the option (through the Python API) to disable telemetry`)

### CODE REVIEW

File: docs/Privacy.md [hunk 1]
```diff
@@ -17,4 +17,5 @@ Currently telemetry is only implemented for Windows builds and is turned **ON**
 The Windows provider uses the [TraceLogging](https://docs.microsoft.com/en-us/windows/win32/tracelogging/trace-logging-about) API for its implementation. This enables ONNX Runtime trace events to be collected by the operating system, and based on user consent, this data may be periodically sent to Microsoft servers following GDPR and privacy regulations for anonymity and data access controls. 
 
 Windows ML and onnxruntime C APIs allow Trace Logging to be turned on/off (see [API pages](../README.md#api-documentation) for details).
-For information on how to enable and disable telemetry, see [C API: Telemetry](./C_API.md#telemetry).
+For information on how to enable and disable telemetry, see [C API: Telemetry](./C_API.md#telemetry). 
+There are equivalent APIs in the C# and Python language bindings as well.
```
Comment (Craigacp) [inline docs/Privacy.md:21]: We've got the same hook in the Java API too -  https://github.com/microsoft/onnxruntime/blob/master/java/src/main/java/ai/onnxruntime/OrtEnvironment.java#L218

Comment (hariharans29) [reply docs/Privacy.md]: Updated the doc to include Java as well, thanks

File: csharp/src/Microsoft.ML.OnnxRuntime/GlobalMethods.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    public static class GlobalMethods
```
Comment (yuslepukhin) [inline csharp/src/Microsoft.ML.OnnxRuntime/GlobalMethods.cs:6]: >ublic static class GlobalMethods [](start = 5, length = 32)

I would recommend to expose this through the Environment class? As Adam says to be consistent with Java?

File: csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.cs [hunk 3]
```diff
@@ -38,41 +38,66 @@ public enum OrtLanguageProjection
     }
 
     /// <summary>
-    /// This class intializes the process-global ONNX runtime
-    /// C# API users do not need to access this, thus kept as internal
+    /// This class initializes the process-global ONNX runtime
     /// </summary>
-    internal sealed class OnnxRuntime : SafeHandle
+    public sealed class OrtEnv : SafeHandle
```
Comment (hariharans29) [inline csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.cs:43]: The class that was previously hosting the native OrtEnv handle is now public sealed OrtEnv

File: csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.cs [hunk 4]
```diff
@@ -38,41 +38,66 @@ public enum OrtLanguageProjection
...
 
-        private OnnxRuntime()  //Problem: it is not possible to pass any option for a Singleton
-            :base(IntPtr.Zero, true)
+        #region public methods
+
+        /// <summary>
+        /// Enable platform telemetry collection where applicable
+        /// (currently only official Windows ORT builds have telemetry collection capabilities)
+        /// </summary>
+        public static void EnableTelemetryEvents()
```
Comment (hariharans29) [inline csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.cs:80]: 2 new public static methods to enable/disable telemetry

File: csharp/src/Microsoft.ML.OnnxRuntime/GlobalMethods.cs (hunk 2 above)
Comment (hariharans29) [reply csharp/src/Microsoft.ML.OnnxRuntime/GlobalMethods.cs]: Done

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**